drivers/flash/Kconfig.stm32

```diff
@@ -31,3 +28,10 @@ config SOC_FLASH_STM32
 	help
 	  Enable STM32F0x, STM32F3x, STM32F4x, STM32F7x, STM32L4x, STM32WBx,
 	  STM32G0x or STM32G4x series flash driver.
+
+config SOC_FLASH_STM32_GENERIC
```

Comment:
Do you have any suggestions?

When reviewing the other implementations, I think it would be trivial (no or minimal code changes) to convert almost all of them to the generic driver. As far as I can tell the only exceptions are the F4 and F7 drivers. The main reason I haven’t done it already is that I don’t have suitable hardware to test on.

Reply:
> Do you have any suggestions?

@andysan let's go with V1, V2, ...

Reply:
OK, so I'll class it `SOC_FLASH_STM32_V1` in that case. Would you be able to test this with the other MCUs affected by this change? I only have access to an L0.

Reply:
@andysan we'll try to test on other mcus this week

Reply:
Renamed the driver to `flash_stm32_v1.c`.